tests/integration/server/regressions/1322-relationship-path-normalization-test.js

```diff
@@ -0,0 +1,73 @@
+import { module, test } from 'qunit';
+import { Model, hasMany, belongsTo, JSONAPISerializer } from 'ember-cli-mirage';
+import Server from 'ember-cli-mirage/server';
+import promiseAjax from 'dummy/tests/helpers/promise-ajax';
+
+module('Integration | Server | Regressions | 1322 Relationship Path Normalization Test', function(hooks) {
+  hooks.beforeEach(function() {
+    this.server = new Server({
+      environment: 'test',
+      models: {
+        happyUser: Model.extend({
+          happyLicenses: hasMany()
+        }),
+        happyLicense: Model.extend({
+          happyUser: belongsTo()
+        })
+      },
+      serializers: {
+        application: JSONAPISerializer.extend({
+          keyForRelationship(relationshipName) {
+            if (relationshipName === 'happyUser') {
+              return 'happy_user';
+            } else {
+              return JSONAPISerializer.prototype.keyForRelationship.apply(this, arguments);
+            }
+          }
+        })
+      },
+      baseConfig() {
+        this.resource('happy-licenses');
+      }
+    });
+  });
+
+  hooks.afterEach(function() {
+    this.server.shutdown();
+  });
+
+  test('it works', async function(assert) {
+    let user1 = this.server.create('happy-user');
+    this.server.create('happy-license', { happyUser: user1 });
+
+    assert.expect(2);
```

Comment:
I believe it doesn't add a value to the test since the only way the assertions wouldn't be executed is a hanging promiseAjax which would lead to the test timeout anyways.